From: Schischu <schischu65@gmail.com>
Date: Sun, 4 May 2014 21:14:05 +0200
Subject: [PATCH] bluedroid: increase uhid report buffer size for wiimote

The wiimote driver > kernel 3.2 uses at least 22bytes, so bluedroid crashes cause it only allows 20bytes max. we increase this limit.

diff --git a/btif/src/btif_hh.c b/btif/src/btif_hh.c
index 59eb5fa..c4bbde2 100644
--- a/btif/src/btif_hh.c
+++ b/btif/src/btif_hh.c
@@ -723,7 +723,7 @@ void btif_hh_disconnect(bt_bdaddr_t *bd_addr)
 void btif_hh_setreport(btif_hh_device_t *p_dev, bthh_report_type_t r_type, UINT16 size,
                             UINT8* report)
 {
-    UINT8  hexbuf[20];
+    UINT8  hexbuf[40];
     UINT16 len = size;
     int i = 0;
     if (p_dev->p_buf != NULL) {
@@ -739,7 +739,7 @@ void btif_hh_setreport(btif_hh_device_t *p_dev, bthh_report_type_t r_type, UINT1
     p_dev->p_buf->offset = BTA_HH_MIN_OFFSET;
 
     //Build a SetReport data buffer
-    memset(hexbuf, 0, 20);
+    memset(hexbuf, 0, 40);
     for(i=0; i<len; i++)
         hexbuf[i] = report[i];
 
